c2corg_ui/static/js/documentediting.js

```diff
@@ -126,22 +126,21 @@ app.DocumentEditingController = function($scope, $element, $attrs,
   // When creating a new document, the model is not created until
   // the form is touched. At least create an empty object.
   this.scope_[this.modelName_] = {};
-
-  if (this.id_ && this.lang_) {
-    if (this.auth_.isAuthenticated()) {
-      // Get document attributes from the API to feed the model:
+  if (this.auth_.isAuthenticated()) {
+    if (this.id_ && this.lang_) {
+     // Get document attributes from the API to feed the model:
       goog.asserts.assert(!goog.isNull(this.id_));
       goog.asserts.assert(!goog.isNull(this.lang_));
       this.api_.readDocument(this.module_, this.id_, this.lang_).then(
           this.successRead_.bind(this)
       );
-    } else {
-      // Redirect to the auth page
-      var current_url = window.location.href;
-      window.location.href = '{authUrl}?from={redirect}'
-          .replace('{authUrl}', this.authUrl_)
-          .replace('{redirect}', encodeURIComponent(current_url));
     }
+  } else {
+    // Redirect to the auth page
+    var current_url = window.location.href;
+    window.location.href = '{authUrl}?from={redirect}'
+        .replace('{authUrl}', this.authUrl_)
+        .replace('{redirect}', encodeURIComponent(current_url));
```

Comment:
"The Javascript session will struggle to continue it's mission in the face
of impending doom. Coding in this manner is considered unpredictable.
...and cruel."
So yeah, let's add a return statement.

On Fri, Feb 26, 2016 at 3:35 PM, Alex Saunier <notifications@github.com>
wrote:

> In c2corg_ui/static/js/documentediting.js
> <https://github.com/c2corg/v6_ui/pull/196#discussion_r54249971>:
>
> >      }
> > +  } else {
> > +    // Redirect to the auth page
> > +    var current_url = window.location.href;
> > +    window.location.href = '{authUrl}?from={redirect}'
> > +        .replace('{authUrl}', this.authUrl_)
> > +        .replace('{redirect}', encodeURIComponent(current_url));
>
> Good catch!
> Would it be useful to add a return here, so that the rest of the function
> is not executed if the redirection is not instantaneous? Or is the function
> already exited when setting window.location.href?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/c2corg/v6_ui/pull/196/files#r54249971>.
>



-- 

*Adam Gina*JavaScript Developer
adam.gina@camptocamp.com

*Camptocamp SA*EPFL Innovation Park
Bâtiment A
1015 Lausanne

Tel + 41 21 619 10 10
http://www.camptocamp.com